dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt

```diff
@@ -332,6 +334,21 @@ internal class CoreFeature(
         serviceName = configuration.service ?: appContext.packageName
         envName = configuration.env
         variant = configuration.variant
+        appBuildId = with(appContext.assets) {
+            try {
+                open(BUILD_ID_FILE_NAME).bufferedReader().use {
+                    it.readText().trim()
+                }
+            } catch (@Suppress("SwallowedException") e: IOException) {
```

Comment:
> [!NOTE]
> We might want to distinguish `FileNotFoundException` from other kinds of `IOException`
> The first might mean that the app was built with an older plugin (or no plugin at all) in which case 
> it's expected to not have a Build ID File. In that case an `INFO` log is indeed enough.
> The second means an actual IO error happened with a file that exists, and this should report at least to `WARNING` or `ERROR` (and maybe be sent to telemetry).

Reply:
Indeed, good point. Makes sense to distinguish between the case when file is not there and when something went wrong during the file read.